### ISSUE TITLE
ID field `displayUsing` works on detail but doesn't work on index

### ISSUE THREAD

**Issue by dmason30** [laravel/nova-issues#5061]
- Nova Version: 4.18.2

### Detailed steps to reproduce the issue on a fresh Nova installation:
```php
ID::make('ID', fn () => '97c06f8d-8bbd-4a98-9a5e-efaa0b0e6bd7')
    ->displayUsing(fn () => 'I am displayUsing value')
    ->showOnPreview(),
```

**Index**
❌ Does not display the displayUsing value
![image](https://user-images.githubusercontent.com/20278756/201967892-983cc9c2-36d1-428b-8572-bea9de7826fb.png)

**Detail**
✅ Displays the text correctly
![image](https://user-images.githubusercontent.com/20278756/201968036-eff99c00-1753-4c24-a213-a595d109b591.png)

### Real world example where this could be useful
I want to display a condensed version of the UUID on index, so would like to be able to do something like this:
```php
ID::make()
    ->displayUsing(fn ($value) => $request->isResourceIndexRequest() ? Str::afterLast($value, '-') : $value)
    ->showOnPreview(),
```
![image](https://user-images.githubusercontent.com/20278756/201968811-953d769b-135f-449f-8c35-b1301fd0d711.png)

**Comment by davidhemphill** [comment]
Just fixed this and will push it out later today. 👍

**Comment by dmason30** [comment]
> Just fixed this and will push it out later today. 👍

@davidhemphill this didn't seem to get fixed in 4.19.1?

**Comment by davidhemphill** [comment]
It's in 4.19.2
